Wurstpack/wurstscript/lib/systems/Knockback3.wurst

```diff
@@ -0,0 +1,217 @@
+    // If a unit's vertical component is not greater than this, it will not
+    // become airborne. Units per second.
+    */
+    static var airborneThreshold = 150.
+
+    /**
+    // For the purposes of friction, distinguish a unit which is airborne
+    // versus one which is sliding. Height in units.
+    */
+    static var isAirborneThreshold = 5.
+
+    /** Below this speed, sliding units will stop. */
+    static var minimumSlideSpeed = 30.
+
+    /** Above this speed, sliding units will spawn a dust effect. */
+    static var frictionFxThreshold = 180.
+    static var frictionFxPath = "Objects\\Spawnmodels\\Undead\\ImpaleTargetDust\\ImpaleTargetDust.mdl"
+
+    /** Above this speed, destructables hit will be destroyed. */
+    static var destroyDestructableSpeedThreshold = 300.
+
+    /** Below this height, destructables hit may be destroyed. */
+    static var destroyDestructableHeightThreshold = 150.
+
+    /** The square rect size used for finding destructables. */
+    @configurable static constant DESTRUCTABLE_ENUM_SIZE    = 130.
+
+    /**
+    // If enabled, units have their move-speed changed while airborne.
+    // Warning: this is not a lock-safe form of crowd control.
+    */
+    @configurable static constant USE_MOVE_SPEED_MODIFIERS  = true
+
+    /**
+    // If enabled, units have their prop window changed while airborne.
+    // Warning: this is not a lock-safe form of crowd control.
+    */
+    @configurable static constant USE_PROP_WINDOW_MODIFIERS = true
+
+
+    /**
+    // Apply a knockback vector to unit u. `velocity` is initial speed in units
+    // per second. `groundAngle` and `airAngle` are the direction and trajectory
+    // parameters, respectively. Example:
+
+    //     import Knocback3
+    //     ...
+    //     unit u = ...
+    //     Knockback3.add(u, 1000., angle(0.), angle(bj_PI / 4.))
+    */
+    static function add(unit u, real velocity, angle groundAngle, angle airAngle)
+        let instVel = velocity*ANIMATION_PERIOD
+        let v       = vec3(0., 0., 0.).polarProject(instVel, groundAngle, airAngle)
+
+        Knockback3 knockback
+        if unitNodes.has(u)
+            knockback      = unitNodes.get(u)
+            knockback.del += v
+        else
+            knockback     = new Knockback3()
+            knockback.u   = u
+            knockback.del = v
+            if size == 1
+                clock.startPeriodic(ANIMATION_PERIOD, function tick)
+
+            if knockback.u.addAbility(HEIGHT_ENABLER)
+                knockback.u.removeAbility(HEIGHT_ENABLER)
+
+
+    static function setVel(unit u, real velocity, angle groundAngle, angle airAngle)
+        if unitNodes.has(u)
+            let knockback = unitNodes.get(u)
+            let instVel   = velocity*ANIMATION_PERIOD
+
+            knockback.del = vec3(0., 0., 0.).polarProject(instVel, groundAngle, airAngle)
+        else
+            add(u, velocity, groundAngle, airAngle)
+
+
+    static function getVel(unit u) returns real
```

Comment:
This is sort of a convenience function for if you want to do something other than add/set the vector. I've never actually used `getVel` in practice. Anyway I've changed this to return the vector.